vars/publishReports.groovy

```diff
@@ -52,11 +56,20 @@ def call(List<String> files, Map params = [:]) {
           "DESTINATION_PATH=${dirname ?: '/'}",
           "PATTERN=${ basename ?: '*' }",
         ]) {
-          // Blob container can be removed once files are uploaded on the azure file storage
-          sh 'az storage blob upload --account-name=prodjenkinsreports --container=reports --timeout=${TIMEOUT} --file=${FILENAME} --name=${FILENAME} ${UPLOADFLAGS} --overwrite'
+          sh '''
+          # If the "failIsEmpty" flag is active and the file is empty, fail the pipeline
+          lineCount=$(wc -l "${filename}" | awk '{print $1}')
+          if [[ "${failIsEmpty}" != false && "${lineCount}" -eq 0 ]]; then
+            echo "ERROR: ${filename} is empty."
+            exit 1
```

Comment:
Fundamental question: do we want the calling pipeline to fail? At first sight it looks like but since the only known use case today is https://github.com/jenkins-infra/helpdesk/issues/4056 then I'm asking.

Is the condition (and its maintenance burden: it is not even tested since it is implemented with shell code) worth the gain considering PHS reports are now always generated?

Reply:
So... rename this PR and change `failIfEmpty` by `skipIfEmpty`?